crates/dmm-tools/src/dmm.rs

```diff
@@ -194,12 +194,15 @@ impl Map {
     }
 
     pub fn adjust_key_length(&mut self) {
-        if self.dictionary.len() > 2704 {
-            self.key_length = 3;
-        } else if self.dictionary.len() > 52 {
-            self.key_length = 2;
-        } else {
-            self.key_length = 1;
+        if let Some(max_key) = self.dictionary.keys().max() {
+            let max_key = max_key.0;
+            if max_key > 2704 {
+                self.key_length = 3;
+            } else if max_key > 52 {
```

Comment:
I think these two conditions should now be >= instead of just >

Reply:
addressed in most recent commit